itou/eligibility/models/geiq.py

```diff
@@ -0,0 +1,243 @@
+
+        return eligibility, amount
+
+    @property
+    def eligibility_confirmed(self) -> bool:
+        """
+        GEIQ elibility.
+        Calculated in function of:
+            - author kind
+            - number, annex and level of administrative criteria.
+        """
+        eligibility, _ = self._get_eligibility_and_allowance_amount()
+
+        return eligibility
+
+    @property
+    def allowance_amount(self) -> int:
+        """
+        Amount of granted allowance for job seeker.
+        Calculated in function of:
+            - author kind
+            - number, annex and level of administrative criteria.
+        Currently, only 3 amounts possible:
+            - 0
+            - 814EUR
+            - 1400EUR
+        """
+
+        _, amount = self._get_eligibility_and_allowance_amount()
+
+        return amount
+
+    @classmethod
+    @transaction.atomic()
+    def create_eligibility_diagnosis(
+        cls,
+        job_seeker: User,
+        author: User,
+        author_structure: Siae | PrescriberOrganization,
+        administrative_criteria=(),
+    ):
+        author_org = author_geiq = author_kind = None
+
+        if isinstance(author_structure, PrescriberOrganization):
+            author_org = author_structure
+            author_kind = AuthorKind.PRESCRIBER
+        elif isinstance(author_structure, Siae) and author_structure.kind == SiaeKind.GEIQ:
+            author_geiq = author_structure
+            author_kind = AuthorKind.GEIQ
+        else:
+            raise ValueError(
+                f"Impossible de créer un diagnostic GEIQ avec une structure de type "
+                f"{author_structure.__class__.__name__}"
+            )
+
+        result = cls.objects.create(
+            job_seeker=job_seeker,
+            author=author,
+            author_kind=author_kind,
+            author_prescriber_organization=author_org,
+            author_geiq=author_geiq,
+        )
+
+        if administrative_criteria:
+            result.administrative_criteria.set(administrative_criteria)
+
+        return result
+
+
+class GEIQAdministrativeCriteria(AbstractAdministrativeCriteria):
+    parent = models.ForeignKey(
+        "self",
+        verbose_name="Critère parent",
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    annex = models.CharField(
+        verbose_name="Annexe",
+        max_length=1,
+        choices=AdministrativeCriteriaAnnex.choices,
+        default=AdministrativeCriteriaAnnex.ANNEX_1,
+    )
+    level = models.CharField(
+        verbose_name="Niveau",
+        max_length=1,
+        choices=AdministrativeCriteriaLevel.choices,
+        # as opposed to IAE, level can be null (annex 1)
+        null=True,
+        blank=True,
+    )
+
+    class Meta:
+        verbose_name = "Critère administratif GEIQ"
+        verbose_name_plural = "Critères administratifs GEIQ"
+        ordering = ["level", "ui_rank"]
```

Comment:
Peut-être grouper les critères sans niveau en dernier ?
```suggestion
        ordering = [F("level").asc(nulls_last=True), "ui_rank"]
```